Design.md

```diff
@@ -0,0 +1,55 @@
+# Design doc for job-runner
```

Comment:
please add a section describing the CLI, including usage examples and flags/arguments

Reply:
Will do